File: sdk/javascript-closure-angular/API/Client/Product.js
goog.provide('API.Client.Product');

/**
 * @record
 */
API.Client.Product = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.Product.prototype.ID;

/**
 * @type {!string}
 * @export
 */
API.Client.Product.prototype.name;

/**
 * @type {!string}
 * @export
 */
API.Client.Product.prototype.description;

/**
 * @type {!number}
 * @export
 */
API.Client.Product.prototype.quantityMultiplier;

/**
 * @type {!number}
 * @export
 */
API.Client.Product.prototype.shipWeight;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Product.prototype.active;

/**
 * @type {!string}
 * @export
 */
API.Client.Product.prototype.type;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Product.prototype.inventoryEnabled;

/**
 * @type {!number}
 * @export
 */
API.Client.Product.prototype.inventoryNotificationPoint;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Product.prototype.variantLevelInventory;

/**
 * @type {!API.Client.Object}
 * @export
 */
API.Client.Product.prototype.xp;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Product.prototype.allowOrderExceedInventory;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Product.prototype.inventoryVisible;

File: sdk/dart/lib/model/product.dart
part of swagger.api;


@Entity()
class Product {
  
  @Property(name: 'ID')
  String ID = null;
  

  @Property(name: 'Name')
  String name = null;
  

  @Property(name: 'Description')
  String description = null;
  

  @Property(name: 'QuantityMultiplier')
  int quantityMultiplier = null;
  

  @Property(name: 'ShipWeight')
  Number shipWeight = null;
  

  @Property(name: 'Active')
  bool active = null;
  

  @Property(name: 'Type')
  String type = null;
  

  @Property(name: 'InventoryEnabled')
  bool inventoryEnabled = null;
  

  @Property(name: 'InventoryNotificationPoint')
  int inventoryNotificationPoint = null;
  

  @Property(name: 'VariantLevelInventory')
  bool variantLevelInventory = null;
  

  @Property(name: 'xp')
  Object xp = null;
  

  @Property(name: 'AllowOrderExceedInventory')
  bool allowOrderExceedInventory = null;
  

  @Property(name: 'InventoryVisible')
  bool inventoryVisible = null;
  
  Product();

  @override
  String toString()  {
    return 'Product[ID=$ID, name=$name, description=$description, quantityMultiplier=$quantityMultiplier, shipWeight=$shipWeight, active=$active, type=$type, inventoryEnabled=$inventoryEnabled, inventoryNotificationPoint=$inventoryNotificationPoint, variantLevelInventory=$variantLevelInventory, xp=$xp, allowOrderExceedInventory=$allowOrderExceedInventory, inventoryVisible=$inventoryVisible, ]';
  }

}

